### PR TITLE
Fix: escape space characters for ftp adapter

### PR DESCRIPTION
**Pull request by RubtsovAV**
Bug example:

```
$filesystem->has('that path/have/the space character.txt');
// Return false every time if in path have space character.
```

Because before call ftp_rawlist we need add a slash in front of each space character in path.

```
$options = '-A';
$path = str_replace(' ', '\ ', $path);
$result = ftp_rawlist($connection, $options . ' ' . $path);
```

### CODE REVIEW

**Comment by RubtsovAV** [comment]
Removed change of Utils.php from the commit.

**Comment by RubtsovAV** [comment]
I not sure how add that feature in the test suite. If you tell me, then I can try do that.

**Comment by RubtsovAV** [comment]
ping

**Comment by sifex** [comment]
Tested. Fixed issue that I was having regarding https://github.com/thephpleague/flysystem/issues/563 and spaces.

**Comment by frankdejonge** [comment]
I'm at a conference now, but: We need tests for this, because the suite currently breaks because of this. So either it should be added in this PR or I can add it myself, but that'd be done when I have time after the conf.

**Comment by barryvdh** [comment]
This breaks on my Linux ftp with dirs with spaces (old versions works there).

Do you have a test Windows FTP account?

**Comment by barryvdh** [comment]
So perhaps use the systemType property to check if it's windows and needs to be escaped?

**Comment by sifex** [comment]
I'm not 100% on this and I may be wrong, but this fix was for my FTPd
Server, maybe this belongs in the FTPd Adapter?

-- 
Alex Sinnott

**Comment by RubtsovAV** [comment]
I think we need create test suite with a docker containers with different ftp servers. Then we will be sure in the adapter on 100%. What do you think?

**Comment by RubtsovAV** [comment]
Write your ftp servers, please.

**Comment by RubtsovAV** [comment]
By now it should work 100% with:
- Pure-FTPd
- vsftpd

**Comment by RubtsovAV** [comment]
@sifex , @barryvdh can you test it on your envorments, pls?

P.s. and if do not complicate, tell me of your ftp server name.

**Comment by sifex** [comment]
@RubtsovAV Sorry, company prohibits it. I'll see what I can do about setting up a ftpd server on my casual box today.

**Comment by RubtsovAV** [comment]
Why PR is not merged?

**Comment by sifex** [comment]
@RubtsovAV Needs to have tests written for it and have the tests to pass.

File: src/Adapter/Ftp.php [hunk 1]
```diff
@@ -502,4 +509,31 @@ public function isConnected()
...
+
+        $response = implode("\n", $response);
+        if (stripos($response, 'Pure-FTPd')) {
+            return 'Pure-FTPd';
+        }
+
+        return null;
+    }
+
+    protected function ftp_rawlist($options, $path)
```
Comment (frankdejonge) [inline src/Adapter/Ftp.php]: This should be named `ftpRawlist`.

**Comment by RubtsovAV** [comment]
Hello?

**Comment by RubtsovAV** [comment]
@frankdejonge Done.

**Pull request merged**